### ISSUE TITLE
docker builds are python2

### ISSUE THREAD

**Issue by adamsteer** [PDAL/PDAL#2402]
Docker images for Alpine and Ubuntu use Python 2, if built from:

https://github.com/PDAL/PDAL/tree/master/scripts/docker/Dockerfile
https://github.com/PDAL/PDAL/blob/master/scripts/docker/alpine/Dockerfile
https://github.com/PDAL/PDAL/blob/master/scripts/docker/ubuntu/Dockerfile

…and should probably be bumped to Python 3.x

The fix seems pretty simple (just rename packages from `py-` to `py3-` or `python-` to `python3-`; and update `libpython-` numbers). Right now I’m having other issues building them locally, happy to make a PR once I can get them going.

...unless someone else is faster; or there are reasons to use a different python version. Noting that the CentOS dockerfile already uses Python 3.4.

**Comment by abellgithub** [comment]
Is there a specific reason that this should be built with Python 3?

**Comment by adamsteer** [comment]
Selfish real reason: I write and communicate and teach in py3.
Rationalisation: py2 end-of-support soon.
Ultimately, I'm happy to maintain a py3-ised PDAL container if need be until the consensus says ‘move’.

I’m still having issues building the containers, I’ll do some more research today.

**Comment by abellgithub** [comment]
I'm not opposed.  I just wondered what the need was.

**Comment by alexgleith** [comment]
Python 2 end of life is next year, 2020: https://legacy.python.org/dev/peps/pep-0373/

In my humble opinion, nobody should be building _only_ Python 2 support, and the PDAL Python library explicity states that it supports Python 3.x, so the Docker builds should too. Really, the Docker builds should _not_ support Python 2, I reckon!

**Comment by alexgleith** [comment]
When I build a downstream Docker Image based on the `pdal/pdal` image, and I install the Python3 library `pdal` I get an error as follows:

```
RUN pip3 install pdal
 ---> Running in d29d75a09583
Collecting pdal
  Downloading https://files.pythonhosted.org/packages/e4/67/338c4164a02b1bb0ecb34adca254b29e88c6c3ed874402d2cd67f5c611a5/PDAL-2.1.8.tar.gz (66kB)
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-ej85l2y9/pdal/setup.py", line 128, in <module>
        raise Exception(message % (built_major, running_major))
    Exception: Version mismatch. PDAL Python support was compiled against version 2.x but setup is running version is 3.x.
```

This is proving pretty hard to conquer...

**Comment by abellgithub** [comment]
PDAL supports python 3, but the docker builds are python 2.  You can certainly build PDAL from source against python 3.  If you want to submit a PR with a Dockerfile for python 3 that would be great.  The alpine image took a bit of doing last time, but may be easier this time around.

**Comment by adamsteer** [comment]
@abellgithub working on that - I hit this: https://github.com/PDAL/PDAL/issues/2278#issuecomment-439722737 in Alpine, and the dockerfile in `scripts/ubuntu` runs into:

```
CMake Error at CMakeLists.txt:168 (find_package):
 Could not find a package configuration file provided by "PDAL" (requested
 version 1.3) with any of the following names:

   PDALConfig.cmake
   pdal-config.cmake

 Add the installation prefix of "PDAL" to CMAKE_PREFIX_PATH or set
 "PDAL_DIR" to a directory containing one of the above files.  If "PDAL"
 provides a separate development package or SDK, be sure it has been
 installed.
```

Will spend a bit of time on both today, but pointers welcome!

**Comment by abellgithub** [comment]
It seems like you're trying to build the python extension or something like that without PDAL being installed in a standard location.  Without more context, I don't know.

**Comment by hobu** [comment]
I have migrated the Ubuntu docker container at `pdal/pdal` to Python 3.

**Comment by abellgithub** [comment]
Fixed in #2630.
